src/merge.c

```diff
@@ -2083,9 +2083,9 @@ static int iterator_for_annotated_commit(
 	opts.flags = GIT_ITERATOR_DONT_IGNORE_CASE;
 
 	if (commit == NULL) {
-		error = git_iterator_for_nothing(out, &opts); 
+		error = git_iterator_for_nothing(out, &opts);
 	} else if (commit->type == GIT_ANNOTATED_COMMIT_VIRTUAL) {
-		error = git_iterator_for_index(out, commit->index, &opts);
+		error = git_iterator_for_index(out, git_index_owner(commit->index), commit->index, &opts);
```

Comment:
I don't think this can be `git_index_owner(commit->index)`, but we could also change `iterator_for_annotated_commit` to take an additional `repo` argument easily.